src/components/Search/Homepage/HomepageSearchAutocomplete.tsx

```diff
@@ -147,9 +147,13 @@ const HomepageSearchAutocomplete: React.FC<{
             }
           }}
           onClose={() => {
-            setIsOpen(false);
-            if (setHideMapToggle) {
-              setHideMapToggle(false);
+            // Only close the dropdown if the user clicks outside the dropdown
+            // (i.e. Element in focus will not have a 'placeholder' attribute).
+            if (!document.activeElement?.hasAttribute('placeholder')) {
+              setIsOpen(false);
+              if (setHideMapToggle) {
+                setHideMapToggle(false);
+              }
```

Comment:
How about giving the input element an ID and use that ID to check if the input element is the active one? I feel like that would be more robust, since the user could click on a different element with a placeholder or the input element might not have one necessarily

```tsx
<div style={{ display: 'flex', alignItems: 'center' }}>
  <StyledTextField
    placeholder="City, county, state, or zip"
    {...params}
    className={searchTextFieldStyles.root}
    $isOpen={isOpen}
    InputProps={{
      ...params.InputProps,
      id={generatedID}
      endAdornment: <SearchBarIcon />,
    }}
  />
  <Hidden smUp>
  <CloseIcon
    onClick={() => setIsOpen(false)}
    $showIcon={isOpen}
  />
  </Hidden>
</div>
// ...

onClose={() => (
  if (document.activeElement?.attribute('id') !== generatedID) {
    // ...
  }
)}

```

Reply:
Unfortunately, if we generate an ID, it will still not override the generated ID from Material UI. This mismatch will prevent us from comparing whether the focused element is the same element which should be open/closed. After some discussion, we decided to merge this PR as is.